enhancements/monitoring/replace-prometheus-adapter-with-metrics-server.md

```diff
@@ -0,0 +1,277 @@
+---
+title: replace-prometheus-adapter-with-metrics-server
+authors:
+  - "@slashpai"
+reviewers:
+  - "@simonpasquier"
+  - "@dgrisonnet"
+  - "@openshift/openshift-team-monitoring"
+approvers: 
+  - "@simonpasquier"
+  - "@dgrisonnet"
+  - "@openshift/openshift-team-monitoring"
+api-approvers:
+  - "@deads2k"
+creation-date: 2023-06-20
+last-updated: 2023-09-25
+tracking-link:
+  - https://issues.redhat.com/browse/MON-3153
+---
+
+# Replace prometheus-adapter with metrics-server
+
+## Summary
+
+This proposal suggests replacing the [Prometheus Adapter](https://github.com/kubernetes-sigs/prometheus-adapter) currently used for implementing the resource [metrics API](https://github.com/kubernetes/metrics) in OpenShift with the [Metrics Server](https://github.com/kubernetes-sigs/metrics-server).
```

Comment:
@slashpai still trying to understand this new stack but the Windows team had added some windows_exporter specific queries to the prometheus adapter: https://github.com/prometheus-operator/kube-prometheus/pull/1058. This was to collect the metrics through the top command, do we need something similar for the metrics-server?

Reply:
Normally no because kubelet has a shim to support windows workload: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/winstats/winstats.go. Kubelet normally relies on cAdvisor to get the metrics, but since it doesn't support windows, there is a dedicated shim to get these metrics. That said, this is still something that we would want to be verified by QE.

Reply:
cc: @juzhao for visibility

Reply:
Thanks we will also test this out when opting-in to this feature.